src/order/complete_lattice.lean

```diff
@@ -70,110 +70,19 @@ notation `⨅` binders `, ` r:(scoped f, infi f) := r
 instance (α) [has_Inf α] : has_Sup αᵒᵈ := ⟨(Inf : set α → α)⟩
 instance (α) [has_Sup α] : has_Inf αᵒᵈ := ⟨(Sup : set α → α)⟩
 
-/--
-Note that we rarely use `complete_semilattice_Sup`
-(in fact, any such object is always a `complete_lattice`, so it's usually best to start there).
+/-- A complete lattice is a bounded lattice which has suprema and infima for every subset.
 
-Nevertheless it is sometimes a useful intermediate step in constructions.
+Note that it suffices to build either the infimum or the supremum for the other to be determined.
+See `complete_lattice_of_Inf` and `complete_lattice_of_Sup`.
 -/
-@[ancestor partial_order has_Sup]
-class complete_semilattice_Sup (α : Type*) extends partial_order α, has_Sup α :=
-(le_Sup : ∀ s, ∀ a ∈ s, a ≤ Sup s)
-(Sup_le : ∀ s a, (∀ b ∈ s, b ≤ a) → Sup s ≤ a)
-
-section
-variables [complete_semilattice_Sup α] {s t : set α} {a b : α}
-
-@[ematch] theorem le_Sup : a ∈ s → a ≤ Sup s := complete_semilattice_Sup.le_Sup s a
-
-theorem Sup_le : (∀ b ∈ s, b ≤ a) → Sup s ≤ a := complete_semilattice_Sup.Sup_le s a
-
-lemma is_lub_Sup (s : set α) : is_lub s (Sup s) := ⟨λ x, le_Sup, λ x, Sup_le⟩
-
-lemma is_lub.Sup_eq (h : is_lub s a) : Sup s = a := (is_lub_Sup s).unique h
-
-theorem le_Sup_of_le (hb : b ∈ s) (h : a ≤ b) : a ≤ Sup s :=
-le_trans h (le_Sup hb)
-
-theorem Sup_le_Sup (h : s ⊆ t) : Sup s ≤ Sup t :=
-(is_lub_Sup s).mono (is_lub_Sup t) h
-
-@[simp] theorem Sup_le_iff : Sup s ≤ a ↔ ∀ b ∈ s, b ≤ a :=
-is_lub_le_iff (is_lub_Sup s)
-
-lemma le_Sup_iff : a ≤ Sup s ↔ ∀ b ∈ upper_bounds s, a ≤ b :=
-⟨λ h b hb, le_trans h (Sup_le hb), λ hb, hb _ (λ x, le_Sup)⟩
-
-lemma le_supr_iff {s : ι → α} : a ≤ supr s ↔ ∀ b, (∀ i, s i ≤ b) → a ≤ b :=
-by simp [supr, le_Sup_iff, upper_bounds]
-
-theorem Sup_le_Sup_of_forall_exists_le (h : ∀ x ∈ s, ∃ y ∈ t, x ≤ y) : Sup s ≤ Sup t :=
-le_Sup_iff.2 $ λ b hb, Sup_le $ λ a ha, let ⟨c, hct, hac⟩ := h a ha in hac.trans (hb hct)
-
--- We will generalize this to conditionally complete lattices in `cSup_singleton`.
-theorem Sup_singleton {a : α} : Sup {a} = a :=
-is_lub_singleton.Sup_eq
-
-end
-
-/--
-Note that we rarely use `complete_semilattice_Inf`
-(in fact, any such object is always a `complete_lattice`, so it's usually best to start there).
-
-Nevertheless it is sometimes a useful intermediate step in constructions.
--/
-@[ancestor partial_order has_Inf]
-class complete_semilattice_Inf (α : Type*) extends partial_order α, has_Inf α :=
+@[protect_proj, ancestor lattice has_bot has_top has_Inf has_Sup]
+class complete_lattice (α : Type*) extends lattice α, has_bot α, has_top α, has_Inf α, has_Sup α :=
```

Comment:
What if we extend `order_bot` and `order_top` here?

Reply:
This is not possible because `bounded_order` is a mixin.